guides/source/configuring.md

```diff
@@ -946,6 +946,48 @@ Specifies if an error should be raised if the order of a query is ignored during
 
 Controls whether migrations are numbered with serial integers or with timestamps. The default is `true`, to use timestamps, which are preferred if there are multiple developers working on the same application.
 
+#### `config.active_record.db_warnings_action`
+
+Controls the action to be taken when a SQL query produces a warning. The following options are available:
+
+  * `:ignore` - Database warnings wil be ignored. This is the default.
+
+  * `:raise` - Database warnings will be raised as `ActiveRecord::SqlWarning`.
+
+  * `:log` - Database warnings will be logged via `ActiveRecord.logger` at the `:warn` level.
+
+  * Custom proc - A custom proc can be provided. It should accept a `SqlWarning` error object.
+
+    For example:
+
+    ```ruby
+    config.active_record.db_warnings_action = ->(warning) do
+      # Report to custom exception reporting service
+      Bugsnag.notify(warning.message) do |notification|
+        notification.add_metadata(:warning_code, warning.code)
+        notification.add_metadata(:warning_level, warning.level)
+      end
+    end
+    ```
+
+This is currently only supported for MySQL adapters and Postgres.
+
+#### `config.active_record.db_warnings_ignore`
+
+Specifies an allowlist of warnings that will be ignored, regardless of the configured `db_warnings_action`.
+The default is `[]`. Warnings to ignore can be specified as Strings or Regexps. For example:
```

Comment:
The empty array is an implementation detail. For docs we should say the following:

```suggestion
The default is behavior is to allow all warnings. Warnings to ignore can be specified as Strings or Regexps. For example:
```